jwst/datamodels/schemas/slitdata.schema.yaml

```diff
@@ -107,6 +107,8 @@ allOf:
     wavelength:
       title: Wavelength array, corrected for zero-point
       fits_hdu: WAVELENGTH
+      ndim: 2
```

Comment:
Do we know whether this may cause any problems downstream in any steps that might be assuming that 3D data will also have a 3D wavelength array? Granted, for TSO modes, the wavelengths in each plane of the 3D stack would be identical, but at some point we'll just need to make sure downstream code isn't trying to access the 3rd dimension of "wavelength".

Reply:
I don't know for sure.  I can't find email about this right now, but my understanding was that we adopted a policy that the wavelength array (attribute) would always be 2-D, regardless of the shape of the SCI data.  For TSO data, a 3-D wavelength array could be large.  The `SlitDataModel` was created relatively recently; it's the one that did not specify ndim and default.  The `CubeModel` schema does already specify "ndim: 2", and "default: 0.0".  Currently, `extract_2d` does not create the wavelength attribute for NRC_TSGRISM data, and `extract_1d` got a 3-D wavelength attribute containing uninitialized memory, so if any step relied on those wavelengths, that step would be using garbage.

Reply:
I think @philhodge is right and not having `ndim` in the schema was an oversight.